tests/ga/test_update.py

```diff
@@ -1892,6 +1895,123 @@ def time(self):
         return current_time
 
 
+class TryUpdateGoalStateTestCase(HttpRequestPredicates, AgentTestCase):
+    """
+    Tests for UpdateHandler._try_update_goal_state()
+    """
+    def test_it_should_return_true_on_success(self):
+        update_handler = get_update_handler()
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            self.assertTrue(update_handler._try_update_goal_state(protocol), "try_update_goal_state should have succeeded")
+
+    def test_it_should_return_false_on_failure(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            def http_get_handler(url, *_, **__):
+                if self.is_goal_state_request(url):
+                    return HttpError('Exception to fake an error retrieving the goal state')
+                return None
+            protocol.set_http_handlers(http_get_handler=http_get_handler)
+
+            update_handler = get_update_handler()
+            self.assertFalse(update_handler._try_update_goal_state(protocol), "try_update_goal_state should have failed")
+
+    def test_it_should_update_the_goal_state(self):
+        update_handler = get_update_handler()
+        with mock_wire_protocol(mockwiredata.DATA_FILE) as protocol:
+            protocol.mock_wire_data.set_incarnation(12345)
+
+            update_handler._try_update_goal_state(protocol)
+
+            self.assertEqual(protocol.get_incarnation(), '12345', "The goal state was not updated (received unexpected incarnation)")
+            self.assertIsNotNone(protocol.client._extensions_goal_state, "The extensions goal state was not updated")  # this is just a dummy test for now
```

Comment:
can this just mock `protocol.client.update_extension_goal_state`? or is that not what this test is testing?

Reply:
It is testing that both the goal state (from wireserver) and the extensions goal state (from the hostgaplugin) are updated by _try_update_goal_state.

sorry, maybe the comment was confusing... by "dummy" I just meant that the test checks for not none, only --- at this point only cares that it got updated with something, even if it is garbage... subsequent PRs will parse the return value and do more meaningful checks